SphericalShell/OpenMC_Output_Processing/SS_Depletion_PostProcessing.py

```diff
@@ -0,0 +1,62 @@
+import openmc
+import openmc.deplete
+import numpy as np
+import matplotlib.pyplot as plt
+
+def extract_nuclides(dep_file_path, time_units, depletable_mat_index) : 
+    '''
+    Identifies nuclides from each depletion timestep and stores them in an array
+    
+    inputs: 
+        dep_file_path : path to .h5 file with OpenMC depletion simulation results
+        time_units : units ('s', 'd', 'min', 'h', 'a') used when retrieving depletion timesteps
+        depletable_mat_index : index of depletable material in OpenMC Materials object
+    '''    
+    dep_results = openmc.deplete.Results(filename=dep_file_path)
+    time_steps = dep_results.get_times(time_units = time_units)
+    nuclide_array = np.ndarray([], dtype = object)
+    stable_init_nuc = np.ndarray([], dtype = object)
+    
+    for step in range(len(time_steps)):
+        # Obtain a materials object with depletion information at each timestep
+        materials_object = dep_results.export_to_materials(step)[depletable_mat_index]
+        all_nuc = materials_object.get_nuclides()      
+        for nuclide in all_nuc :   
+            # Remove stable nuclides at beginning of operation
+            if step == 0:
+                half_life = openmc.data.half_life(nuclide)
+                if half_life == None:
+                    stable_init_nuc = np.append(stable_init_nuc, nuclide)
+            if nuclide not in nuclide_array and nuclide not in stable_init_nuc:        

```

Comment:
You might consider using a `set()` for this rather than a numpy array.  With sets I would propose that you simply build a set of all nuclides - you don't need to check wats already in the set because sets do that automatically. You can also build a set of all the stable nuclides at the initial time.  Then, at the end, you can remove the set of stables from the set of all nuclides.  The code will probably be much simpler and straightforward.